packages/jest-core/src/SearchSource.ts

```diff
@@ -51,13 +51,22 @@ const toTests = (context: Context, tests: Array<Config.Path>) =>
     path,
   }));
 
+const hasSCM = (changedFilesInfo: ChangedFiles) => {
+  const {repos} = changedFilesInfo;
+  // no SCM (git/hg/...) is found in any of the roots.
+  const noSCM = Object.values(repos).every(scm => scm.size === 0);
+  return !noSCM;
+};
+
 export default class SearchSource {
   private _context: Context;
+  private _dependencyResolver: DependencyResolver | null;
   private _testPathCases: TestPathCases = [];
 
   constructor(context: Context) {
     const {config} = context;
     this._context = context;
+    this._dependencyResolver = null;
```

Comment:
I think we may not immediately use `DependencyResolver` in `SearchSource`, so at first `this._dependencyResolver` will be `null`. When we need `DependencyResolver`, just call `_getDependencyResolver` to create/get it.